crowdstrike/crowdstrike/openbas_crowdstrike.py

```diff
@@ -0,0 +1,234 @@
+from datetime import datetime, timedelta
+
+import pytz
+from crowdstrike.crowdstrike_api_handler import CrowdstrikeApiHandler
+from crowdstrike.query_strategy.alert import Alert
+from crowdstrike.query_strategy.base import Base
+from dateutil.parser import parse
+from pyobas.helpers import (
+    OpenBASCollectorHelper,
+    OpenBASConfigHelper,
+    OpenBASDetectionHelper,
+)
+from pyobas.signatures.signature_type import SignatureType
+from pyobas.signatures.types import MatchTypes, SignatureTypes
+
+
+class OpenBASCrowdStrike:
+    def __init__(
+        self,
+        strategy: Base,
+        config: OpenBASConfigHelper,
+        helper: OpenBASCollectorHelper,
+        detection_helper: OpenBASDetectionHelper,
+        signature_types,
+    ):
+        self.strategy = strategy
+        self.config = config
+        self.helper = helper
+        self.detection_helper = detection_helper
+        self.signature_types = signature_types
+
+    def _fetch_expectations(self, start_time):
+        self.helper.collector_logger.info("Gathering expectations for executed injects")
+        expectations = (
+            self.helper.api.inject_expectation.expectations_assets_for_source(
+                self.config.get_conf("collector_id")
+            )
+        )
+        self.helper.collector_logger.info(
+            "Found " + str(len(expectations)) + " expectations waiting to be matched"
+        )
+
+        valid_expectations = []
+
+        for expectation in expectations:
+            # Parse the creation date of the expectation
+            expectation_date = parse(
+                expectation["inject_expectation_created_at"]
+            ).astimezone(pytz.UTC)
+
+            # Check if the expectation is expired: created date is greater than request start time
+            if expectation_date < start_time:
+                self.helper.collector_logger.info(
+                    f"Expectation expired, failing inject {expectation['inject_expectation_inject']} "
+                    f"({expectation['inject_expectation_type']})"
+                )
+                self.helper.api.inject_expectation.update(
+                    expectation["inject_expectation_id"],
+                    {
+                        "collector_id": self.config.get_conf("collector_id"),
+                        "result": (
+                            "Not Detected"
+                            if expectation["inject_expectation_type"] == "DETECTION"
+                            else "Not Prevented"
+                        ),
```

Comment:
Agreed, however to achieve this we need to create common components and it's probably outside of the scope of this feature by now... I created #75 to consider this.

As it's not a big change, it could be sneaked into another feature some time.